p2p/enode/iter.go

```diff
@@ -62,7 +62,7 @@ func CycleNodes(nodes []*Node) Iterator {
 }
 
 type sliceIter struct {
-	mu    sync.Mutex
+	mu    sync.RWMutex
```

Comment:
I think RWMutex is not necessary for this. Please change it back to sync.Mutex.

Reply:
done